sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -66,6 +66,11 @@ public void AddIntegerAttribute(string name, int value)
             _activityAdapter?.AddTag(name, value);
         }
 
+        public void AddLongAttribute(string name, long value)
```

Comment:
Couldn't this be done by just calling the existing method:

```csharp
scope.AddAttribute<long>("somename", 123456L, (val, _) => val);
```

Maybe we should consider just making this an add-without-format?

```csharp
 public void AddAttribute<T>(string name, T value) =>
    AddAttribute<T>(name, value,  (val, _) => val);
```

Reply:
Cool.  I didn't realize there was prior art that this was based on.